pkg/operator/api/api.go

```diff
@@ -449,7 +493,11 @@ func (a *apiServer) ListResiliency(ctx context.Context, in *operatorv1pb.ListRes
 // TODO: @joshvanl: Authorize pod name and namespace matches the SPIFFE ID of
 // the caller.
 func (a *apiServer) ComponentUpdate(in *operatorv1pb.ComponentUpdateRequest, srv operatorv1pb.Operator_ComponentUpdateServer) error { //nolint:nosnakecase
-	if err := a.authzRequest(srv.Context(), in.GetNamespace()); err != nil {
+	spiffeID, err := getSpiffeIDFromContext(srv.Context())
```

Comment:
Filtering must also be done here, else applications will receive the Components which they are not scoped to when they are updated..

Reply:
I have updated it to load only components, NOT meant for control-plane.
And, do not want to add ComponentUpdate feature to control plane components as of now.

Reply:
We gain no security posture by not extending this check to also `ComponentUpdate` as control plane/non-control plane components will be available to any client which calls it and the component has some informer event.

Can you please describe the purpose of having any distinction at all between control plane components and app IDs in the scope entity list? It is still not clear to me why we would not do a simple filter based on the entities inside the scope, rather than the (in my eyes) surprising behaviour of special casing known control plane services and blocking any app ID from having access to it?

Reply:
I have updated to do ComponentUpdate for both CP and non-CP. I have kept the same logic as is currently for ListComponents.